lib/bloc/account/account_bloc.dart

```diff
@@ -231,6 +238,68 @@ class AccountBloc {
     action.resolve(await _breezLib.sendWalletCoins(action.destAddress, Int64(action.feeRate)));     
   }
 
+  Future _exportPaymentsAction(ExportPayments action) async {
+    action.resolve(await _exportPayments());
+  }
+
+  Future _exportPayments() async {
+    // convert payment list to csv format and save to disk
+    return await _saveCsvFile(const ListToCsvConverter().convert(_generatePaymentList()));
+  }
+
+  _generatePaymentList(){
+    List<List<dynamic>> paymentList = new List.generate(_filterPayments(_paymentsController.value.paymentsList).length, (index) {
+      List paymentItem = new List();
+      paymentItem.add(DateUtils.formatYearMonthDayHourMinute(DateTime.fromMillisecondsSinceEpoch(
+          _filterPayments(_paymentsController.value.paymentsList).elementAt(index).creationTimestamp.toInt() * 1000)));
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).title);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).description);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).destination);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).amount.toString());
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).preimage);
+      paymentItem.add(_filterPayments(_paymentsController.value.paymentsList).elementAt(index).paymentHash);
+      return paymentItem;
+    });
+    paymentList.insert(0, ["Date & Time", "Title", "Description", "Node ID", "Amount", "Preimage", "TX Hash"]);
+    return paymentList;
+  }
+
+  Future _saveCsvFile(String csv) async {
+    try {
+      String filePath = await _createCsvFilePath();
+      final file = File(filePath);
+      await file.writeAsString(csv);
+      return file.path;
+    } catch (e) {
+      return e;
```

Comment:
No need for try/catch here. In case of exception the Future will emit error.

Reply:
[Removed](https://github.com/breez/breezmobile/pull/203/commits/9283c8287b1e67abe92fa083e41d65f66be6842d)